src/viam/sdk/common/linear_algebra.hpp

```diff
@@ -5,18 +5,22 @@
 #include <boost/qvm/vec.hpp>
 #include <boost/qvm/vec_traits.hpp>
 
-#include <viam/api/common/v1/common.pb.h>
+#include <viam/sdk/common/proto_convert.hpp>
+
+VIAM_SDK_API_FWD_NAMESPACE_BEGIN(common)
```

Comment:
I'm open to getting rid of the macro, it's more of a convenience than anything, and also makes it easy to grep for where forward declarations are happening.

Putting everything in a single `fwd` header is doable, but I think that from a user's perspective I prefer that
- if you do want to escape hatch into the proto conversions, the same header with the SDK types you're working with also shows you the relevant API types
- you don't have to introduce all the other forward decls into scope

Reply:
OK, those are good enough arguments against the fwd header that I think we can eliminate that.

How about a compromise? We keep the macro for now, but file a ticket that when the SDK goes to C++17, the macro gets removed, because then it becomes a lot less painful to write:

```
namespace ::viam::api::common::v1 {
class Vector3;
}
```

So, a ticket for that labeled as C++17? That or just do it explicitly now with C++14 namespaces. I've got a pretty strong aversion to macros in public headers if at all avoidable.

Also, it is probably worth filing a ticket now, if one doesn't exist, to introduce prelude/postlude headers to do macro push/pops so that the tooling for avoiding macro leakage is in place. It will be required in order to support adding ABI visibility annotations and/or calling convention annotations should we ever want to support windows.